2018-gophercon-iceland/ml-on-go.md

```diff
@@ -0,0 +1,42 @@
+# Machine Learning On Go Code
+
+**Updates**:  https://github.com/src-d/conferences/issues/7
+**Sent on**:  2018/03/03
+**Status**:   proposed
+**Author**:   Francesc Campoy
+**Slides**:   TBA
+**Proposal**: https://www.papercall.io/talks/64650/children/64651
+
+**Abstract**:
+
+We've all wondered how to use Machine Learning with Go, but what about turning the tables for once? What can Machine Learning do *for* Go?
+
+In this talk we will discover how different Machine Learning models can help us write better go by predicting from our next character to our next bug!
+
+**Description**:
+
+Machine Learning is definitely what the cool kids are doing nowadays. Many wonder how to implement simple Machine Learning models, such as [image recognition](https://outcrawl.com/image-recognition-api-go-tensorflow/), and many frameworks are popping out such as [Tensorflow for Go](https://godoc.org/github.com/tensorflow/tensorflow/tensorflow/go), [GoLearn](https://github.com/sjwhitworth/golearn), or [Gorgonia](https://github.com/gorgonia/gorgonia).
+
+What not many are talking about is how to those techniques to improve our developer routines. Machine Learning on Source Code (MLonCode) is a very interesting field because it is at the frontier of [Natural Language Processing](https://en.wikipedia.org/wiki/Natural-language_processing), [Graph-Based Machine Learning](https://research.googleblog.com/2016/10/graph-powered-machine-learning-at-google.html), Static Analysis, and has the power to even bring other fields like Dynamic Analysis of programs.
+
+The amount of data available for this problem is almost overwhelming, and given that data is the fuel of Machine Learning, we are excited for an amazing ride!
+
+This talk will cover the basics of what Machine Learning techniques can be applied to source code, specifically we will discover:
+
+- [embeddings over identifiers](https://blog.sourced.tech/post/id2vec/),
+- structural embeddings over source code, answering the question how similar are two fragments of code,
+- recurrent neural networks for code completion,
+- future direction of the research.
```

Comment:
Once I finish the Code Neuron (analogy to Sentiment Neuron), it can be a funny problem to mention. The idea is to detect the code blocks inside natural text applying the same DL pipeline. Early experiments show that it is indeed possible; I will have to provide the working end-to-end solution by March 16th.  It backs to Tensorflow, and we could try to run it on TPUs to make it even awesomer. The current work in progress is https://github.com/vmarkovtsev/codeneuron (even no README so far, but the code already works). Trained on StackOverflow, applied to golang-nuts.

We can mention topic modeling - we had good results in 2016, and the same core will likely to be used for snippets exploratory search.

Another option is to talk about how seriously we took the code deduplication task - in the end, it is not only data mining but also hyperparameter optimization with ML based on human feedback (I mean code-annotation app here).

Reply:
Nice, I'll keep this comment here and see whether this talk gets accepted or not.